controller/src/main/java/io/pravega/controller/task/Stream/StreamMetadataTasks.java

```diff
@@ -626,11 +628,18 @@ public void setRequestEventWriter(EventStreamWriter<ControllerEvent> requestEven
                                 .thenCompose(y -> {
                                     final OperationContext context = streamMetadataStore.createContext(scope, stream);
 
-                                    return withRetries(() ->
-                                            streamMetadataStore.addUpdateStreamForAutoStreamCut(scope, stream,
-                                                    config.getRetentionPolicy(), context, executor)
-                                            .thenCompose(v ->  streamMetadataStore.setState(scope, stream, State.ACTIVE,
-                                                    context, executor)), executor)
+                                    return withRetries(() -> {
```

Comment:
The change overall looks good. One high-level comment I have is that I find in general a bad idea to use `null` to indicate the absence of something. In this case, "something" is a retention policy. Could we instead use a constant that represents the absence of a policy in a stream configuration? We would need to make changes to the client to do this.

Reply:
I absolutely agree. Should i open a different issue for it or address it here?

Reply:
Ok.

Reply:
opened issue https://github.com/pravega/pravega/issues/2167